privacyidea/api/lib/utils.py

```diff
@@ -247,19 +250,46 @@ def verify_auth_token(auth_token, required_role=None):
     :param required_role: list of "user" and "admin"
     :return: dict with authtype, realm, rights, role, username, exp, nonce
     """
+    r = None
     if required_role is None:
         required_role = ["admin", "user"]
     if auth_token is None:
         raise AuthError(_("Authentication failure. Missing Authorization header."),
                         id=ERROR.AUTHENTICATE_AUTH_HEADER)
-    try:
-        r = jwt.decode(auth_token, current_app.secret_key, algorithms=['HS256'])
-    except jwt.DecodeError as err:
-        raise AuthError(_("Authentication failure. Error during decoding your token: {0!s}").format(err),
-                        id=ERROR.AUTHENTICATE_DECODING_ERROR)
-    except jwt.ExpiredSignature as err:
-        raise AuthError(_("Authentication failure. Your token has expired: {0!s}").format(err),
-                        id=ERROR.AUTHENTICATE_TOKEN_EXPIRED)
+
+    headers = jwt.get_unverified_header(auth_token)
+    algorithm = headers.get("alg")
+    if algorithm in TRUSTED_JWT_ALGOS:
+        # The trusted JWTs are RSA, PSS or eliptic curve signed
+        trusted_jwts = current_app.config.get("PI_TRUSTED_JWT", [])
+        for trusted_jwt in trusted_jwts:
+            try:
+                if trusted_jwt.get("algorithm") in TRUSTED_JWT_ALGOS:
```

Comment:
Do we still want to restrict the required algorithm in the trusted JWT description? Technically we just need the public key and the list of allowed algorithms in `jwt.decode()`. That way, the *creator* of the JWT can change the algorithm (as long as it is in the trusted algorithms list).

Reply:
￼	Yes. Merge it I explain on Monday.